src/app/ReadHandler.cpp

```diff
@@ -373,6 +379,8 @@ CHIP_ERROR ReadHandler::ProcessReadRequest(System::PacketBufferHandle && aPayloa
     }
     ReturnErrorOnFailure(err);
 
+    VerifyOrReturnError(InteractionModelEngine::GetInstance()->CheckResourceQuotaForCurrentFabric(this), CHIP_ERROR_NO_MEMORY);
```

Comment:
The biggest issue is we cannot shutdown a subscription which is now reporting data gracefully since it is waiting for the response of the report data and the exchange context (CRMP infact) today only supports one ReliableMessageContext per exchange context.